packages/backend/src/index.ts

```diff
@@ -160,6 +168,20 @@ async function main() {
     .addRouter('', metricsHandler())
     .addRouter('/api', apiRouter)
     .addRouter('', await app(appEnv));
+  backendPlugins.map(containerizedPlugin => {
+    const container = rootApplicationContext.getChildContext(
+      containerizedPlugin.name,
+    );
+    containerizedPlugin.getDependencies().forEach(dependency => {
+      try {
+        container.get(dependency);
+      } catch (e) {
+        const s = `Failed to retrieve injected dependency for ${dependency}`;
+        logger.error(s);
```

Comment:
Instead of logging and throwing, can this be `throw new CustomErrorBase(`failed to retrieve injected dependency for ${dependency}`, e)` with `CustomErrorBase` replaced by some new child of that error? This current implementation with log & throw will print two separate log lines for the same error.

Reply:
There's [ForwardedError](https://github.com/backstage/backstage/blob/5cd8efd5766ebcc47115a485196abc41d0d18881/packages/errors/src/errors/common.ts#L85) if there's nothing more specific.

Reply:
I added a new subtype but this is something that needs to be revisited after we start polishing this. There's no error handling elsewhere yet.